#
# Copyright 2019 Giuseppe Silano, University of Sannio in Benevento, Italy
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# Sphinx.msg modified in 2024 by Pascal Goldschmid, University of Stuttgart, Germany
#
# This is custom message developed to bring data coming from the Sphinx data logger.
# All measurements are expressed in the SI (meter, radians, etc.)
#
float64 velXABC                      # Velocity along the x-axis in the aircraft body center reference system
float64 velYABC                      # Velocity along the y-axis in the aircraft body center reference system
float64 velZABC                      # Velocity along the z-axis in the aircraft body center reference system
float64 angVelXABC                   # Angular velocity along the x-axis in the aircraft body center reference system
float64 angVelYABC                   # Angular velocity along the y-axis in the aircraft body center reference system
float64 angVelZABC                   # Angular velocity along the z-axis in the aircraft body center reference system
float64 velXENU                      # Velocity along the x-axis in the inertial reference system
float64 velYENU                      # Velocity along the y-axis in the inertial reference system
float64 velZENU                      # Velocity along the z-axis in the inertial reference system
float64 posX                         # Drone position along the x-axis
float64 posY                         # Drone position along the y-axis
float64 posZ                         # Drone position along the z-axis
float64 attitudeX                    # Drone attitude along the x-axis (roll angle)
float64 attitudeY                    # Drone attitude along the y-axis (pitch angle)
float64 attitudeZ                    # Drone attitude along the z-axis (yaw angle)
float32 timeStamp                    # Drone time stamp (seconds)
float64 accXABC                      # Acceleration along the x-axis in the aircraft body center reference system
float64 accYABC                      # Acceleration along the y-axis in the aircraft body center reference system
float64 accZABC                      # Acceleration along the z-axis in the aircraft body center reference system
float64 angAccXABC                   # Angular acceleration along the x-axis in the aircraft body center reference system
float64 angAccYABC                   # Angular acceleration along the y-axis in the aircraft body center reference system
float64 angAccZABC                   # Angular acceleration along the z-axis in the aircraft body center reference system


#Add camera position  data
float64 gimbalPitchMotorAngle
float64 gimbalRollMotorAngle